ecosystem/typescript/sdk_v2/src/core/account.ts

```diff
@@ -0,0 +1,168 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import nacl from "tweetnacl";
+import * as bip39 from "@scure/bip39";
+import { AccountAddress } from "./account_address";
+import { Hex } from "./hex";
+import { bytesToHex } from "@noble/hashes/utils";
+import { HexInput } from "../types";
+import { PrivateKey, PublicKey, Signature } from "../crypto/asymmetric_crypto";
+import { derivePath } from "../utils/hd-key";
+import { AuthenticationKey } from "../crypto/authentication_key";
+import { Ed25519PrivateKey, Ed25519PublicKey, Ed25519Signature } from "../crypto/ed25519";
+
+/**
+ * Class for creating and managing account on Aptos network
+ *
+ * Use this class to create accounts, sign transactions, and more.
+ * Note: Creating an account instance does not create the account onchain.
+ */
+export class Account {
+  /**
+   * A private key and public key, associated with the given account
+   */
+  readonly publicKey: PublicKey;
+  readonly privateKey: PrivateKey;
+
+  /**
+   * Account address associated with the account
+   */
+  readonly accountAddress: AccountAddress;
+
+  /**
+   * constructor for Account
+   *
+   * TODO: This constructor uses the nacl library directly, which only works with ed25519 keys.
+   * Need to update this to use the new crypto library if new schemes are added.
+   *
+   * @param args.privateKey PrivateKey - private key of the account
+   * @param args.address AccountAddress - address of the account
+   *
+   * This method is private because it should only be called by the factory static methods.
+   * @returns Account
+   */
+  private constructor(args: { privateKey: PrivateKey; address: AccountAddress }) {
```

Comment:
I am thinking of modifying this to 

`private constructor(args: { privateKey: PrivateKey; publicKey: PublicKey; address: AccountAddress })`

So we don't have to worry about schemes. All properties will just get assigned directly.

Reply:
We should probably add a PrivateKey -> PublicKey function on the PrivateKey.  The great thing about asymmetrical key cryptography, is the public key should be derivable from the private key.

You will want to add an optional publicKey though for MultiEd25519

Reply:
If this is multisig account, then we probably don't have private key and cannot derive the public.

Maybe I will just have a optional privateKey in this case.

`private constructor(args: { privateKey?: PrivateKey; publicKey: PublicKey; address: AccountAddress })`